codec-http/src/main/java/io/netty/handler/codec/http/HttpResponseDecoder.java

```diff
@@ -123,39 +135,80 @@ public HttpResponseDecoder(
         super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED);
     }
 
+    /**
+     * @deprecated Prefer the {@link #HttpResponseDecoder(int, int, int)} constructor,
+     * to always enable header validation.
+     */
+    @Deprecated
     public HttpResponseDecoder(
             int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, boolean validateHeaders) {
         super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, validateHeaders);
     }
 
+    /**
+     * @deprecated Prefer the {@link #HttpResponseDecoder(int, int, int, int)} constructor,
+     * to always enable header validation.
+     */
+    @Deprecated
     public HttpResponseDecoder(
             int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, boolean validateHeaders,
             int initialBufferSize) {
         super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, validateHeaders,
               initialBufferSize);
     }
 
+    public HttpResponseDecoder(
+            int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, int initialBufferSize) {
+        super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, DEFAULT_VALIDATE_HEADERS,
+              initialBufferSize);
+    }
+
+    /**
+     * @deprecated Prefer the {@link #HttpResponseDecoder(int, int, int, int, boolean)} constructor,
+     * to always enable header validation.
+     */
+    @Deprecated
     public HttpResponseDecoder(
             int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, boolean validateHeaders,
```

Comment:
Same true at this level, users need a way to control name/value validation separately. We can consider other options, like a builder you proposed above or maybe instead of taking boolean flags we can take functions to create `Default[Full]HttpResponse`/`DefaultLastHttpContent`